registry/endpoint.go

```diff
@@ -227,6 +227,21 @@ func (e *Endpoint) pingV2() (RegistryInfo, error) {
 	}
 	defer resp.Body.Close()
 
+	// The endpoint may have multiple supported versions.
+	// Ensure it supports the v2 Registry API.
+	var supportsV2 bool
+
+	for _, versionName := range resp.Header[http.CanonicalHeaderKey("Docker-Distribution-API-Version")] {
```

Comment:
https://groups.google.com/forum/#!topic/golang-nuts/gVgVLQU1FFE

On Tue, Jan 20, 2015 at 3:40 PM, Josh Hawn <notifications@github.com> wrote:

> In registry/endpoint.go
> <https://github.com/docker/docker/pull/10227#discussion_r23266959>:
>
> > @@ -227,6 +227,21 @@ func (e *Endpoint) pingV2() (RegistryInfo, error) {
> >  	}
> >  	defer resp.Body.Close()
> >
> > +	// The endpoint may have multiple supported versions.
> > +	// Ensure it supports the v2 Registry API.
> > +	var supportsV2 bool
> > +
> > +	for _, versionName := range resp.Header[http.CanonicalHeaderKey("Docker-Distribution-API-Version")] {
>
> it will range over a nil slice, which is safe, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/10227/files#r23266959>.
>